src/Mvc/Mvc.TagHelpers/src/LinkTagHelper.cs

```diff
@@ -520,18 +523,71 @@ private void BuildLinkTag(string href, TagHelperAttributeList attributes, TagHel
 
     private void AppendVersionedHref(string hrefName, string hrefValue, TagHelperContent builder)
     {
-        if (AppendVersion == true)
-        {
-            hrefValue = FileVersionProvider.AddFileVersionToPath(ViewContext.HttpContext.Request.PathBase, hrefValue);
-        }
-
+        hrefValue = GetVersionedResourceUrl(hrefValue);
         builder
             .AppendHtml(hrefName)
             .AppendHtml("=\"")
             .Append(hrefValue)
             .AppendHtml("\" ");
     }
 
+    private string GetVersionedResourceUrl(string url)
```

Comment:
Is there a way this logic can be de-duped with the method of the same name in `ImageTagHelper`?

Reply:
Same for `ScriptTagHelper`

Reply:
Done, shared the logic in an internal static helper method.